runtime/src/installed_scheduler_pool.rs

```diff
@@ -391,11 +486,72 @@ impl BankWithScheduler {
     }
 
     pub const fn no_scheduler_available() -> InstalledSchedulerRwLock {
-        RwLock::new(None)
+        RwLock::new(SchedulerStatus::Unavailable)
     }
 }
 
 impl BankWithSchedulerInner {
+    fn with_active_scheduler(
+        self: &Arc<Self>,
+        f: impl FnOnce(&InstalledSchedulerBox) -> ScheduleResult,
+    ) -> ScheduleResult {
+        let scheduler = self.scheduler.read().unwrap();
+        match &*scheduler {
+            SchedulerStatus::Active(scheduler) => {
+                // This is the fast path, needing single read-lock most of time.
+                f(scheduler)
+            }
+            SchedulerStatus::Stale(_pool, (result, _timings)) if result.is_err() => {
+                Err(SchedulerAborted)
+            }
+            SchedulerStatus::Stale(_pool, _result_with_timings) => {
+                drop(scheduler);
+
+                let context = SchedulingContext::new(self.bank.clone());
+                let mut scheduler = self.scheduler.write().unwrap();
+                scheduler.transition_from_stale_to_active(|scheduler_pool, result_with_timings| {
+                    scheduler_pool.register_timeout_listener(self.do_create_timeout_listener());
+                    scheduler_pool.take_resumed_scheduler(context, result_with_timings)
+                });
+                drop(scheduler);
+
+                self.with_active_scheduler(f)
+            }
+            SchedulerStatus::Unavailable => panic!(),
```

Comment:
again should have an error message here.

`Unavailable` seems to be used for more than just transitions (at first glance), so I am still trying to convince myself this cannot happen.

Reply:
done: 06b7bff84588eef78752831a845d8d2762803804